src/mrnet/network/reaction_network.py

```diff
@@ -1572,294 +1577,312 @@ def mols_w_cuttoff(RN_pr_solved, cutoff=0, build_pruned_network=True):
-                                    p1 = None
-                                PR = out_node.split(",")[0].split("+PR_")[1]
-                                p2 = out_node.split(",")[1]
-                                if p1 is None:
-                                    reactant1 = int(start)
-                                    reactant2 = int(PR)
-                                    product1 = int(p2)
-                                    product2 = None
-                                    glist = [reactant1, reactant2, product1, product2]
-
-                                else:
-                                    reactant1 = int(start)
-                                    reactant2 = int(PR)
-                                    product1 = int(p1)
-                                    product2 = int(p2)
-                                    glist = [reactant1, reactant2, product1, product2]
-
-                                if set(glist).issubset(set(mols_to_keep)) and (
-                                    {reactant1, reactant2} != {product1, product2}
+                                    node_str not in RN.graph.nodes
+                                    and set(glist).issubset(set(mols_to_keep))
+                                    and set(combined_reactants)
+                                    != set(combined_products)
                                 ):
-                                    count = count + 1
-                                    # print(glist, set(glist).issubset(set(mols_to_keep)))
-                                    reactions.append(
-                                        (
-                                            [reactant1, reactant2],
-                                            [product1, product2],
+                                    unique_reactions.append(
+                                        [combined_reactants, combined_products]
+                                    )
+                                    reactions_with_in_out_nodes.append(
+                                        [
+                                            combined_reactants,
+                                            combined_products,
                                             [in_node, out_node],
-                                        )
+                                        ]
                                     )
-                count_total = count_total + count
-                # print(node, entry.formula, count)
-        print("identify_concerted_rxns_via_intermediates end", time.time())
-        print("total number of unique concerted reactions:", count_total)
 
-        return reactions
+        # print(node, "unique reactions", len(unique_reactions))
+        return unique_reactions, reactions_with_in_out_nodes
 
     @staticmethod
-    def add_concerted_rxns(full_network_pr_solved, pruned_network_build, reactions):
+    def add_concerted_rxns(RN, reactions, temp=-2.15):
```

Comment:
input name "temp" feels incorrect here. What do we name the electron free energy elsewhere? Also, that input parameter isn't described in the doc string. Finally, would there be a way to just pull that info from the RN we pass? It feels weird that the elementary network could be built with one value of electron free energy and then concerteds could be added with a different value.